src/workers/parser/mapBindings.js

```diff
@@ -80,18 +107,22 @@ export default function mapExpressionBindings(
         return replaceNode(ancestors, newNode);
       }
 
-      if (t.isPattern(node.left)) {
-        shouldUpdate = false;
-        return;
-      }
+      return;
     }
 
-    if (!t.isVariableDeclaration(node)) {
+    if (t.isObjectPattern(node)) {
```

Comment:
As mentioned above, I think we'd want to handle these explicitly in `globalizeAssignment` and `globalizeDeclaration` rather than relying on traversal to reach them.